shell/scl_source

```diff
@@ -7,7 +7,7 @@ Don't use this script outside of SCL scriptlets!
 Options:
     -h, --help    display this help and exit"
 
-if [ $# -eq 0 -o $1 = "-h" -o $1 = "--help" ]; then
+if [ $# -eq 0 -o X"$1" = X"-h" -o X"$1" = X"--help" ]; then
```

Comment:
Command line utilities usually don't return non-zero exit status when --help is requested.  OTOH help2man e.g. also accepts non-zero exit status ... yeah, and this is not to be used interactively anyway, so non-zero return makes sense.

AFAIK, the 'X' hack is not needed unless you write really portable scripts (which we are not because we would use `test` probably ...and /bin/sh).  Can you please double-quote only and squash into one commit?  (btw., there is like dozen of similar issues which I plan to fix via separate PR, unless you will be faster and fix it via this PR). :)